### DISCUSSION TITLE
Problem during build process

### DISCUSSION THREAD

**Discussion by Arnon-E** [zeux/pugixml#398] [Q&A] [answered]
Hi,

I add the 3 src files (pugiconfig.hpp, pugixml.cpp, pugixml.hpp) to my application and while building I got an error on pugixml.hpp(614): error C2064: term does not evaluate to a function taking 1 arguments
It's inside the function:
```
// Find attribute using predicate. Returns first attribute for which predicate returned true.
		template <typename Predicate> xml_attribute find_attribute(Predicate pred) const
		{
			if (!_root) return xml_attribute();

			for (xml_attribute attrib = first_attribute(); attrib; attrib = attrib.next_attribute())
				if (pred(attrib))
					return attrib;

			return xml_attribute();
		}

```

in line: `if (pred(attrib))`

Why it's happening and Can any one help me with it?

**Comment by brandl-muc** [comment, marked as answer]
Make sure that the predicate `pred` with which you instantiated your function template `find_attribute()` actually is a function or functor/lambda taking exactly one argument. 

Most likely this is not the case and the compiler does not know how it should call the predicate with a single argument.

**Reply by Arnon-E** [reply]
10x for your help
